### DISCUSSION TITLE
Don't stop serving endpoint when proxy source lost

### DISCUSSION THREAD

**Discussion by matthew-jones-uk** [bluenviron/mediamtx#751] [Questions]
I'm using the proxy feature with home assistant to proxy from a CCTV camera. The connection to the camera is occasionally a bit shaky and drops out momentarily, causing the proxy to bounce the path. 
Home Assistant doesn't play very nicely with this, and often fails to reconnect and just loops the video instead of resuming once the stream is back up.
Is there anyway to continue serving the endpoint, and simply switch to showing a still image, the last frame, or even a blank screen? Thanks.

**Comment by aler9** [comment]
> Is there anyway to continue serving the endpoint, and simply switch to showing a still image, the last frame, or even a blank screen? Thanks.

This is unfortunately technically impossible, since a publisher is free to change:
* track count and kind
* the video codec
* the frame rate
* the video size

Furthermore this would require re-encoding the stream.

Therefore, the only way for a server to support reconnection is to disconnect all readers when a camera disconnects.

[a comment by an automated account is not shown]
